### PR TITLE
Cleanup tests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -354,7 +354,12 @@ public function getProjectDir()
     {
         if (null === $this->projectDir) {
             $r = new \ReflectionObject($this);
-            $dir = $rootDir = \dirname($r->getFileName());
+
+            if (!file_exists($dir = $r->getFileName())) {
+                throw new \LogicException(sprintf('Cannot auto-detect project dir for kernel of class "%s".', $r->name));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Kernel.php]: this missing check made us write bad test cases, fixed below

File: src/Symfony/Component/HttpKernel/Tests/KernelTest.php [hunk 2]
```diff
@@ -980,10 +980,11 @@ protected function getKernel(array $methods = [], array $bundles = [])
 
     protected function getKernelForTest(array $methods = [], $debug = false)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: this method is only used by legacy tests and should be removed when merging up

File: src/Symfony/Component/Lock/Tests/Store/BlockingStoreTestTrait.php [hunk 3]
```diff
@@ -30,6 +30,9 @@ trait BlockingStoreTestTrait
      *
      * This test is time sensible: the $clockDelay could be adjust.
      *
+     * It also fails when run with the global ./phpunit test suite.
+     * @group transient
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Lock/Tests/Store/BlockingStoreTestTrait.php]: /cc @jderusse FYI

File: src/Symfony/Component/Serializer/Tests/Encoder/ChainEncoderTest.php [hunk 4]
```diff
@@ -84,42 +84,13 @@ public function testNeedsNormalizationBasic()
         $this->assertTrue($this->chainEncoder->needsNormalization(self::FORMAT_2));
     }
 
-    /**
-     * @dataProvider booleanProvider
-     */
-    public function testNeedsNormalizationChainNormalizationAware($bool)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Tests/Encoder/ChainEncoderTest.php]: This case relies on `ChainNormalizationAwareEncoder`, which extends `ChainEncoder`, but the class is `@final` so cannot be extended. A fix on phpunit-bridge is coming on branch 4.4 to trigger the missing deprecation.

File: src/Symfony/Component/Lock/Tests/Store/BlockingStoreTestTrait.php (hunk 3 above)
Comment (jderusse) [reply src/Symfony/Component/Lock/Tests/Store/BlockingStoreTestTrait.php]: last time we had random failure, Cache component emptied the Memcached server wich removes locks and failed the test. Are you sure we don't have conflict with other component here?

**Pull request merged**